mindmeld/components/dialogue.py

```diff
@@ -1226,11 +1251,11 @@ def to_json(instance):
         serialized_obj = {}
         for attribute, value in vars(instance).items():
             if isinstance(value, (Params, Request, FrozenParams)):
-                serialized_obj[attribute] = DialogueResponder.to_json(value)
-            elif isinstance(value, tuple) and all(
+                serialized_obj[attribute] = value.to_dict()
+            elif isinstance(value, tuple) and any(
                 isinstance(item, immutables.Map) for item in value
             ):
-                serialized_obj[attribute] = tuple(dict(item) for item in value)
+                serialized_obj[attribute] = [dict(item) for item in value]
```

Comment:
This should resolve the history issue. There's no nesting in history beyond this level, right?

Reply:
Yes, I have guaranteed the history will be a list of responders